spec/classes/agent_spec.rb

```diff
@@ -160,4 +161,76 @@
       it { should contain_file('/etc/zabbix/zabbix_agentd.conf').with_content %r{^TLSPSKFile=/etc/zabbix/keys/tlspskfile.key$} }
     end
   end
+
+  context 'On Ubuntu' do
+    context 'trusty' do
+      let :facts do
```

Comment:
rspec-facts is a cool tool which parses the metadata.json and generates a hash of facts for each OS version in the metadata.json.  I already implemented that for a few spec tests: https://github.com/voxpupuli/puppet-zabbix/blob/master/spec/classes/sender_spec.rb#L7-L13

you can always join the #voxpupuli IRC channel on freenode if you've question. direct chat will make it easier for both of us - we're always happy to help.